proto/storage/signature_integration.proto

```diff
@@ -19,3 +21,28 @@ message CosignPublicKeyVerification {
   }
   repeated PublicKey public_keys = 3;
 }
+
+// Holds all verification data for verifying certificates attached to cosign signatures.
+// In case both the certificate and chain are given, both will be used to verify the public key
+// as well as the chain.
+// If only the certificate is given, the Fulcio trusted root chain will be assumed and verified against.
+// If only the chain is given, this will be used over the Fulcio trusted root chain for verification.
+// If no certificate or chain is given, the Fulcio trusted root chain will be assumed and verified against.
+message CosignCertificateVerification {
+  // PEM encoded certificate chain to use for verification.
+  string certificate_chains_pem_enc = 1;
+  // PEM encoded certificate to use for verification.
+  string certificate_pem_enc = 2;
+  // Certificate OIDC issuer to verify against.
+  // This supports regular expressions following the RE2 syntax: https://github.com/google/re2/wiki/Syntax.
+  // In case the certificate does not specify an OIDC issuer, you may use '.*' as the OIDC issuer. However,
+  // it is recommended to use Fulcio compatible certificates according to the specification:
+  // https://github.com/sigstore/fulcio/blob/main/docs/certificate-specification.md.
```

Comment:
Should we pin the links?

Reply:
I've kept them floating for the time being since this will be within our API docs, and having a fixed version may be outdated in the future in case things are changed in the spec which do not affect cert verification. Also, things have been somewhat stable for the spec, so I'd just leave it tbh.